impeller/renderer/context.h

```diff
@@ -155,6 +156,22 @@ class Context {
   ///
   virtual std::shared_ptr<CommandBuffer> CreateCommandBuffer() const = 0;
 
+  using ReadbackCallback =
+      std::function<void(std::shared_ptr<const DeviceBuffer> buffer)>;
```

Comment:
Should this API emit a constant copy of the host byte data rather than a `DeviceBuffer`? I believe later we we can remove the unnecessary blit on all of the backends and even support onscreen textures on GLES.

Reply:
> I believe later we we can remove the unnecessary blit on all of the backends and even support onscreen textures on GLES.

What?

Reply:
Sorry, I meant the device->device copy, as blitting to a `DeviceBuffer` implies.

Reply:
Ahh yeah makes sense. I went back and forth a few times on what the API should be. This actually needs to end up as. a unique_ptr of something, so that I can use the Dart typed data APIs for readback (eventually). So If I change this to be something like std::unique_ptr<const uint8_t[]> or whatever, I don't know C++ , that should all work nicely right?

Reply:
Yeah, we could also just return an `std::vector<uint8_t>` and just use move semantics to avoid copies when it makes sense, especially since we're getting pretty good help from clang tidy in that department at this point.

Reply:
This kind of thing would jive well with a non-DeviceBuffer variant of `AddCopy` I mentioned [above](https://github.com/flutter/engine/pull/46394#discussion_r1341809385). Might as well eat an extra copy on the host for now and fix both the copies later. :)